ledger/shelley/genesis_test.go

```diff
@@ -250,3 +254,161 @@ func TestGenesisUtxos(t *testing.T) {
+                    "margin": 0.0,
+                    "pledge": 0,
+                    "publicKey": "0aedc455785463235311c990f68742c9043cd79af09ab31c2ba5e195",
+                    "vrf": "eb53a17fbad9b7ea0bcf1e1ea89355305600d593b426dfc3084a924d8877d47e",
+                    "rewardAccount": {
+                        "credential": {
+                            "key hash": "6079cde665c2035b8d9ac8929307bdd7f20a51e678e9d4a5e39ace3a"
+                        },
+                        "network": "Testnet"
+                    }
+                }
+            },
+            "stake": {
+                "24632b71152f31516054075897d0d4ababc33204f8a8661136d49e36": "0aedc455785463235311c990f68742c9043cd79af09ab31c2ba5e195"
+            }
+        },
+        "protocolParams": {
+            "minFeeA": 44,
+            "minFeeB": 155381,
+            "maxBlockBodySize": 65536,
+            "maxTxSize": 16384,
+            "maxBlockHeaderSize": 1100,
+            "keyDeposit": 2000000,
+            "poolDeposit": 500000000
+        }
+    }`
+
+	t.Run("TestInitialPools", func(t *testing.T) {
+		genesis, err := shelley.NewShelleyGenesisFromReader(strings.NewReader(testGenesis))
+		if err != nil {
+			t.Fatalf("Genesis parsing failed: %v", err)
+		}
+
+		pools, delegators, err := genesis.InitialPools()
+		if err != nil {
+			t.Fatalf("InitialPools failed: %v", err)
+		}
+
+		expectedPoolId := "0aedc455785463235311c990f68742c9043cd79af09ab31c2ba5e195"
+		expectedStakeKey := "24632b71152f31516054075897d0d4ababc33204f8a8661136d49e36"
+
+		// Test pool count
+		if len(pools) != 1 {
+			t.Errorf("Expected 1 pool, got %d", len(pools))
+		}
+
+		// Test pool data
+		pool, exists := pools[expectedPoolId]
+		if !exists {
+			t.Fatal("Expected pool not found")
+		}
+
+		if pool.Cost != 340000000 {
+			t.Errorf("Expected pool cost 340000000, got %d", pool.Cost)
+		}
+
+		// Test delegators
+		if len(delegators) != 1 {
+			t.Errorf("Expected 1 delegator mapping, got %d", len(delegators))
+		}
+
+		delegs := delegators[expectedPoolId]
+		if len(delegs) != 1 {
+			t.Errorf("Expected 1 delegator, got %d", len(delegs))
+		} else {
+			// Verify address format
+			addrBytes, err := delegs[0].Bytes()
+			if err != nil {
+				t.Fatalf("Failed to get address bytes: %v", err)
+			}
+
+			// Should be 29 bytes (1 header + 28 stake key)
+			if len(addrBytes) != 29 {
+				t.Errorf("Expected address length 29, got %d", len(addrBytes))
+			}
+
+			// Verify testnet script stake address header
+			// In TestInitialPools and TestPoolById, replace the header check with:
+			if addrBytes[0] != expectedTestnetScriptStakeHeader {
+				t.Errorf("Expected header byte %x, got %x",
+					expectedTestnetScriptStakeHeader, addrBytes[0])
+			}
+
+			// Verify stake key matches
+			stakeKey := hex.EncodeToString(addrBytes[1:])
+			if stakeKey != expectedStakeKey {
+				t.Errorf("Delegator key mismatch:\nExpected: %s\nActual:   %s",
+					expectedStakeKey, stakeKey)
+			}
```

Comment:
Instead of operating on address bytes above, use the `.NetworkId()` and `.Type()` functions from the `Address` type to verify the correct header flags, and the `.StakeKeyHash()` function to get the stake key for comparison.